### ISSUE TITLE
When grepping (`search rg`), preview could highlight the first match

### ISSUE THREAD

**Issue by mikavilpas** [sxyazi/yazi#1095]
### Please describe the problem you're trying to solve

When I search using S (`search rg`), and preview a file that matched my search, the file preview could
- scroll the preview to a section that shows the location of the first match, so that I can see the context of the match
- highlight the first match, so that I can quickly see where it matched

### Would you be willing to contribute this feature?

- [X] Yes, I'll give it a shot

### Describe the solution you'd like

This is the neovim + telescope implementation

<img width="1719" alt="image" src="https://github.com/sxyazi/yazi/assets/300791/8bcdf1ba-801a-4085-beea-c43c346db78e">

The matched files are displayed on the left, and the preview on the right.

Can it be done the same way? I think it's enough to just display each file once, even if it had multiple matches.

### Additional context

Not sure what the performance implications might be.

**Comment by DreamMaoMao** [comment]
This task is want to make like [fg yazi] (https://github.com/DreamMaoMao/fg.yazi) to  a built-in function?

**Comment by mikavilpas** [comment]
Hmm, conceptually they are very similar, but here are some differences:

- fg.yazi seems to use external programs - I actually meant simple highlighting in yazi's current search
- fg.yazi seems capable of showing each match as a separate result - yazi shows each matched file only once

**Comment by soderluk** [comment]
+1 for this. It's a bit annoying to have to open the file, then search again, when I can't see where the matches are in the preview.

**Comment by sxyazi** [comment]
Closing in favor of https://github.com/sxyazi/yazi/issues/3346
